swarm-ui/src/lib/components/ui/loader.svelte

```diff
@@ -0,0 +1,59 @@
+<script lang="ts">
+	import type { HTMLAttributes } from 'svelte/elements'
+	type Dimension = 'default' | 'large' | 'compact' | 'small'
+	type Color = 'high' | 'low'
+	type Props = {
+		dimension?: Dimension
+		color?: Color
+	}
+	let { dimension = 'default', color = 'low' }: Props & HTMLAttributes<HTMLDivElement> = $props()
+
+	let size = dimensionToSize(dimension)
+
+	function dimensionToSize(dimension: Dimension): number {
```

Comment:
This could make it even better and used for icons as they often have specific size variants.
```suggestion
	function dimensionToSize(dimension: Dimension): 32 | 24 | 16 {
```